runtime/lua/vim/lsp/log.lua

```diff
@@ -59,7 +83,7 @@ do
         if arg == nil then
           table.insert(parts, "nil")
         else
-          table.insert(parts, vim.inspect(arg, {newline=''}))
```

Comment:
I sneaked in a behavior change here as I prefer a more structured output, e.g.:

```
[ DEBUG ] 2020-04-16T15:36:57Z+0200 ] ...rdi/.config/nvim/pack/plugins/start/nvim-dap/lua/dap.lua:147 ]	{
  body = {
    scopes = { {
        expensive = false,
        name = "Locals",
        source = vim.empty_dict(),
        variablesReference = 2
      } }
  },
  command = "scopes",
  request_seq = 30,
  seq = 30,
  success = true,
  type = "response"
}
[ DEBUG ] 2020-04-16T15:36:57Z+0200 ] ...rdi/.config/nvim/pack/plugins/start/nvim-dap/lua/dap.lua:224 ]	{
  arguments = {
    variablesReference = 2
  },
  command = "variables",
  seq = 31,
  type = "request"
}
```

I suspect this could be controversial. Maybe I add a `format_func` argument to `create_logger` to make this configurable?

Reply:
Yes, format func or similar would be good. Don't want to force people to have their log files get much bigger just due to pretty printing the log. Can be registered per logger.